latency/src/bin/z_pong.rs

```diff
@@ -32,35 +35,45 @@ async fn main() {
     env_logger::init();
 
     // Parse the args
-    let opt = Opt::from_args();
+    let opt = Opt::parse();
 
-    let mut config = Properties::default();
-    config.insert("mode".to_string(), opt.mode.clone());
+    let mut config = Config::default();
 
-    config.insert("multicast_scouting".to_string(), "false".to_string());
     match opt.mode.as_str() {
-        "peer" => config.insert("listener".to_string(), opt.locator),
-        "client" => config.insert("peer".to_string(), opt.locator),
+        "peer" => {
+            config.set_mode(Some(WhatAmI::Peer)).unwrap();
+            config
+                .listeners
+                .extend(opt.locator.split(',').map(|v| v.parse().unwrap()));
+        }
+        "client" => {
+            config.set_mode(Some(WhatAmI::Client)).unwrap();
+            config
+                .peers
+                .extend(opt.locator.split(',').map(|v| v.parse().unwrap()));
+        }
         _ => panic!("Unsupported mode: {}", opt.mode),
     };
 
-    let zenoh = Zenoh::new(config.into()).await.unwrap();
-    let workspace = zenoh.workspace(None).await.unwrap();
-    let mut sub = workspace
-        .subscribe(&"/test/ping/".to_string().try_into().unwrap())
-        .await
-        .unwrap();
+    config.scouting.multicast.set_enabled(Some(false)).unwrap();
 
-    while let Some(change) = sub.next().await {
-        match change.value.unwrap() {
-            Value::Raw(_, payload) => {
-                workspace
-                    .put(&"/test/pong".try_into().unwrap(), payload.into())
-                    .wait()
-                    .unwrap();
-            }
-            _ => panic!("Invalid value"),
-        }
+    let session = zenoh::open(config).await.unwrap();
+
+    // The resource to echo the data back
+    let reskey_pong = session.declare_expr("/test/pong").await.unwrap();
```

Comment:
As describe above, let's consider `declare*()` with another command-line option in another PR.